### PR TITLE
Plugins Catalog: Only allow admins to access plugins catalog

### LINKED ISSUE #54927: Plugins Catalog: Only admins should access the plugins route `/plugins`
**What happened**:

https://user-images.githubusercontent.com/1173299/189175473-694b639a-2deb-4b6b-9133-5266ccbfd009.mov

**What you expected to happen**:
Should either 404 or redirect to home (which is the same behaviour as `/datasources`)

**How to reproduce it (as minimally and precisely as possible)**:
Log in as Org Viewer (non Grafana admin) and visit `/plugins` by punching it in manually into the address bar

**Anything else we need to know?**:
Related to https://github.com/grafana/grafana/issues/54926, but it seems like the Correlations tab probably shouldn't be here or at the vert least, not error out.

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
Currently any user can access the plugins-catalog routes by accessing the URL directly. This PR adds roles to the frontend route definitions to only allow admins (org and server) to view the catalog.

**Which issue(s) this PR fixes**:

Fixes #54927

**Special notes for your reviewer**:
I'm not sure I'm doing this correctly... maybe I should define more fine grained access control with RBAC?

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM!

**Review by leventebalogh — APPROVED** [review]
Nice 👍

File: public/app/features/plugins/admin/routes.tsx [hunk 1]
```diff
@@ -10,18 +10,21 @@ const DEFAULT_ROUTES = [
   {
     path: '/plugins',
     navId: 'plugins',
+    roles: () => ['Admin'],
```
Comment (gamab) [inline public/app/features/plugins/admin/routes.tsx:13]: I think the legacy access-control is slightly more complex than this, the plugins NavLink is displayed with:
`rc.OrgRole == org.RoleAdmin || cfg.PluginAdminEnabled && rc.IsGrafanaAdmin`

Someone with `Grafana Admin` rights not being the current `Org Admin` cannot access the NavLink any more (I agree this is a strange use case :sweat_smile:)
edge case

**Review by academo — APPROVED** [review]

**Review by gamab — COMMENTED** [review]
Hey thanks for tackling this issue, as of now, I would say don't worry about RBAC related permissions, we are lacking a plugins permission that would simplify the checks.

As I commented, there is a weird edge case with someone being: `Grafana Admin`+`Viewer`, not sure what the best option is here, @wbrowne do you have an opinion? Maybe we can just ignore such a weird case...

However, to be safer, I would inline the backend routes access-control here:
https://github.com/grafana/grafana/blob/8949e574e6288671343b80991479e2e2b9adfc88/pkg/api/api.go#L116-L119
Like this:
```go
	r.Get("/plugins", plugins.ReqCanAdminPlugins(hs.Cfg), hs.Index)
	r.Get("/plugins/:id/", plugins.ReqCanAdminPlugins(hs.Cfg), hs.Index)
	r.Get("/plugins/:id/edit", plugins.ReqCanAdminPlugins(hs.Cfg), hs.Index) // deprecated
	r.Get("/plugins/:id/page/:page", plugins.ReqCanAdminPlugins(hs.Cfg), hs.Index)
```

File: public/app/features/plugins/admin/routes.tsx (hunk 1 above)
Comment (jackw) [reply public/app/features/plugins/admin/routes.tsx]: @gamab Thanks for pointing this out. This [commit](https://github.com/grafana/grafana/pull/57101/commits/d0b540617089c5b195b01d2ee71991e5e5b3e3ae) should fix the issue.

**Review by gamab — APPROVED** [review]
LGTM :)

File: public/app/features/plugins/admin/routes.tsx (hunk 1 above)
Comment (gamab) [reply public/app/features/plugins/admin/routes.tsx]: Thanks!

**Pull request merged**